uaclient/log.py

```diff
@@ -61,6 +62,17 @@ def format(self, record: logging.LogRecord) -> str:
         return json.dumps(list(local_log_record.values()))
 
 
+def get_user_or_root_log_file_path() -> str:
+    """
+    Gets the correct log_file path,
+    adjusting for whether the user is root or not.
+    """
+    if util.we_are_currently_root():
+        return UAConfig().log_file
+    else:
+        return "~/.cache/" + defaults.USER_CACHE_SUBDIR + "/ubuntu-pro.log"
```

Comment:
Since we only call this when non-root, let's re-use the function below
```suggestion
        return get_user_log_file()
```